data-model/src/store.rs

```diff
@@ -102,6 +102,44 @@ pub enum PayloadAppendError {
 /// Returned when no entry was found for some criteria.
 pub struct NoSuchEntryError();
 
+/// Orderings for a
+pub enum QueryOrder {
+    /// Ordered by subspace, then path, then timestamp
+    Subspace,
+    /// Ordered by path, them timestamp, then subspace
+    Path,
+    /// Ordered by timestamp, then subspace, then path
+    Timestamp,
+    /// Whichever order is most efficient.
+    Efficient,
+}
+
+/// An event which took place within a [`Store`].
+/// Each event includes a *progress ID* which can be used to *resume* a subscription at any point in the future.
+pub enum StoreEvent<const MCL: usize, const MCC: usize, const MPL: usize, N, S, PD, AT>
+where
+    N: NamespaceId,
+    S: SubspaceId,
+    PD: PayloadDigest,
+    AT: AuthorisationToken<MCL, MCC, MPL, N, S, PD>,
+{
+    /// A new entry was ingested.
+    Ingested(u64, AuthorisedEntry<MCL, MCC, MPL, N, S, PD, AT>),
+    /// An existing entry received a portion of its corresponding payload.
+    Appended(u64, LengthyAuthorisedEntry<MCL, MCC, MPL, N, S, PD, AT>),
+    /// An entry was forgotten.
+    EntryForgotten(u64, AuthorisedEntry<MCL, MCC, MPL, N, S, PD, AT>),
+    /// A payload was forgotten.
+    PayloadForgotten(u64, PD),
+    /// An entry was overwritten.
+    Overwritten(u64, AuthorisedEntry<MCL, MCC, MPL, N, S, PD, AT>),
+    /// An entry was pruned via prefix pruning.
+    Pruned(u64, AuthorisedEntry<MCL, MCC, MPL, N, S, PD, AT>),
+}
+
+/// Returned when the store could not resume a subscription because the provided ID was too outdated or not present.
```

Comment:
```suggestion
/// Returned when the store chooses to not resume a subscription.
```

I feel like stores should be completely free in which subscriptions to resume and which to refuse. No need for us to specify specific circumstances in which they need to do either.